### ISSUE TITLE
Fix S4830 FN: Struct support is missing

### ISSUE THREAD

**Issue by dmitry-golovinov-sonarsource** [SonarSource/sonar-dotnet#4404]
### Description

S4830 rule does not support method declared in structs

### Repro steps

```
struct AssignmentStruct
    {
        HttpWebRequest CreateRQ()
        {
            return (HttpWebRequest)System.Net.HttpWebRequest.Create("http://localhost");
        }

        public void InitAsArgument(RemoteCertificateValidationCallback callback)
        {
            CreateRQ().ServerCertificateValidationCallback += callback;  //FN, should be Non-compliant
        }

        static void Execute()
        {
            new AssignmentStruct().InitAsArgument((sender, certificate, chain, SslPolicyErrors) => true);  //should be a secondary location
        }
    }
```

### Expected behavior

Issue with a corresponding secondary location should be raised here.

### Actual behavior

Issue is not being raised.

### Related information

* C#/VB.NET Plugins version 8.22
